src/Language/Parser.php

```diff
@@ -248,7 +258,39 @@ public static function __callStatic(string $name, array $arguments) : Node
     {
         $parser = new Parser(...$arguments);
         $parser->expect(Token::SOF);
-        $type = $parser->{'parse' . $name}();
+
+        if (in_array(
+            $name,
+            [
+                'arguments',
+                'valueLiteral',
+                'array',
+                'object',
+                'objectField',
+                'directives',
+                'directive',
+            ],
+            true
+        )) {
+            $type = $parser->{'parse' . $name}(false);
+        } elseif ($name === 'constArguments') {
```

Comment:
Is this critical code? I wonder if it might be more performant to use a map...

Reply:
> Is this critical code?

Not at all, it's backs a bunch of obscure convenience functions for some users that manipulate schemas on the SDL level.

> I wonder if it might be more performant to use a map...

Changed to use a switch, fits better and should be optimal in terms of performance.

Reply:
> should be optimal in terms of performance

How do you figure? I know some languages will optimize by converting a switch statement to a lookup table, but I don't know if PHP is that smart. If it isn't, then you still have to 14 string operations to get down to the default case. But if the code isn't critical, then I suppose there's no point worry about it.

Reply:
I guess converting to actual methods will be a much bigger performance boost than anything we could do in a magic function.

Reply:
Well, I was wondering if a map of keys to callbacks might be more efficient. Something like:

```php

$lookup = [
  'constArguments' => 'parseArguments`,
  'constValueLiteral` => 'parseValueLiteral`,
  ...
];

$type = $parser->{$lookup[$name] ?? 'parse' . $name}(true);

```

Untested. It's only an idea.

Reply:
I prefer not to spend any more time pondering possible microoptimizations to this marginally and non-performance critical bit of functionality. Feel free to add such in a future PR.

Reply:
You're the boss 👍